lhotse/recipes/voxceleb.py

```diff
@@ -0,0 +1,333 @@
+of both audio and video, comprising over 2000 hours of speech. Each segment is at least 
+3 seconds long.
+
+The dataset consists of two versions, VoxCeleb1 and VoxCeleb2. Each version has it's own 
+train/test split. For each version, the YouTube URLs, face detections and tracks, audio files, 
+cropped face videos and speaker meta-data are provided. There is no overlap between the 
+two versions.
+
+- VoxCeleb1: VoxCeleb1 contains over 100,000 utterances for 1,251 celebrities.
+  http://www.robots.ox.ac.uk/~vgg/data/voxceleb/
+- VoxCeleb2: VoxCeleb2 contains over a million utterances for 6,112 identities.
+  http://www.robots.ox.ac.uk/~vgg/data/voxceleb2/ 
+
+LICENSE: The VoxCeleb dataset is available to download for commercial/research purposes 
+under a Creative Commons Attribution 4.0 International License. The copyright remains with 
+the original owners of the video.
+
+This Lhotse recipe prepares the VoxCeleb1 and VoxCeleb2 datasets.
+"""
+import logging
+from pathlib import Path
+from typing import Dict, Optional, Tuple, Union
+from collections import defaultdict, namedtuple
+
+from concurrent.futures.thread import ThreadPoolExecutor
+from tqdm.auto import tqdm
+
+from lhotse import (
+    MonoCut,
+    CutSet,
+    Recording,
+    RecordingSet,
+    SupervisionSegment,
+    SupervisionSet,
+)
+from lhotse.utils import Pathlike, urlretrieve_progress
+from lhotse.qa import validate_recordings_and_supervisions
+from lhotse.manipulation import combine
+
+VOXCELEB1_TRIALS_URL = "http://www.openslr.org/resources/49/voxceleb1_test_v2.txt"
+
+SpeakerMetadata = namedtuple(
+    "SpeakerMetadata", ["id", "name", "gender", "nationality", "split"]
+)
+
+
+def prepare_voxceleb(
+    voxceleb1_root: Optional[Pathlike] = None,
+    voxceleb2_root: Optional[Pathlike] = None,
+    output_dir: Optional[Pathlike] = None,
+    num_jobs: int = 1,
+) -> Dict[str, Dict[str, Union[RecordingSet, SupervisionSet]]]:
+    """
+    Prepare manifests for the VoxCeleb v1 and v2 corpora.
+
+    The manifests are created in a dict with three splits: train, dev and test, for each
+    of the two versions.
+    Each split contains a RecordingSet and SupervisionSet in a dict under keys 'recordings' and 'supervisions'.
+
+    :param voxceleb1_root: Path to the VoxCeleb v1 dataset.
+    :param voxceleb2_root: Path to the VoxCeleb v2 dataset.
+    :param output_dir: Path to the output directory.
+    :param num_jobs: Number of parallel jobs to run.
+    :return: A dict with standard corpus splits ("train" and "test") containing the manifests.
+
+    NOTE: We prepare the data using the Kaldi style split, i.e., the whole VoxCeleb2
+    ("dev" and "test") and the training portion ("dev") of VoxCeleb1 are put into the
+    "train" split. The "test" split contains the "test" portion of VoxCeleb1. So if
+    VoxCeleb1 is not provided, no "test" split is created in the output manifests.
+
+    Example usage:
```

Comment:
To make it render well with sphinx, add double colon, a newline between the header and code, and an indentation for the code.